server/src/plugins/tx-monitor.ts

```diff
@@ -0,0 +1,69 @@
+import fp from 'fastify-plugin';
+import { FastifyInstance } from 'fastify';
+import { IncomingMessage, ServerResponse, Server } from 'http';
+
+declare module 'fastify' {
+  export interface FastifyInstance<
+    HttpServer = Server,
+    HttpRequest = IncomingMessage,
+    HttpResponse = ServerResponse
+  > {
+    authenticate: any;
+  }
+}
+
+const cron = require('node-cron');
+const fetch = require('node-fetch');
+
+import { getPendingTxs, updateTransactionStatus } from '../db';
+import { TransactionStatus } from '../types';
+
+const EXPLORER_API_BASE = 'https://api.blockcypher.com/v1';
+
+export default fp(function transactionsMonitorCron(
+  fastify: FastifyInstance<Server, IncomingMessage, ServerResponse>,
+  opts,
+  next
+) {
+    const monitor = async () => {
+        const pendingTransactions = await getPendingTxs();
+        if (!pendingTransactions || pendingTransactions.length === 0) return;
+        ​
+        const txPromises = pendingTransactions.map(async ({ tx_hash: txHash }) => {
+            const json: any = await getTxStatus(txHash);
+        ​
+            if (json) {
+                if (json.confirmed) {
+                    await updateTransactionStatus(txHash, TransactionStatus.passed);
+                } else if (json.execution_error) {
+                    await updateTransactionStatus(txHash, TransactionStatus.failed);
+                }
+            }
+        ​
+            return json;
+        });
+        ​
+        const results = Promise.all(txPromises);
+        return results;
+    };
+      ​
+
+  fastify.decorate('updateTransactions', async () => {
+    cron.schedule('*/12 * * * * *', () => monitor());
```

Comment:
Optionally you can define `monitor` directly without the callback if you prefer a cleaner syntax.

Code example:
```typescript
cron.schedule('*/12 * * * * *', monitor);
```